### PR TITLE
Reading username, password, and cloud from environmental variables and config files

### PR DESCRIPTION
**Pull request by anneCarlson**
I still need to update the readme and do more thorough testing. Most testing so far has been manual as I'm not certain the best way to test this.

### CODE REVIEW

**Comment by anneCarlson** [comment]
@madisonmay Would be great to get a code review on this. I think it's pretty thoroughly and the README is updated.

**Comment by anneCarlson** [comment]
Just realized I need to update imageapis to work for this as well. On it!

**Comment by anneCarlson** [comment]
Updated image apis and require all calls to authenticate.

File: src/test/java/io/indico/api/test/unit/configTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    // A version of the Config class where all environment variables are whatever
+    // you send to the constructor
+    private static class MockConfigLogic extends ConfigLogic {
+        private String environVariable;
+
+        public MockConfigLogic(String envVars) {
+            environVariable = envVars;
+        }
+
+        @Override
```
Comment (madisonmay) [inline src/test/java/io/indico/api/test/unit/configTest.java]: A real quick note as to why this is necessary would be nice.

File: src/test/java/io/indico/api/test/unit/configTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    @Test
+    public void testCloud() throws Throwable {
+        try {
+            ClassLoader classLoader = getClass().getClassLoader();
+            URL testIni = null;
+            File config;
+
+            MockConfigLogic configLogic = new MockConfigLogic("indico-test-env");
```
Comment (madisonmay) [inline src/test/java/io/indico/api/test/unit/configTest.java]: Another quick comment here since this is a bit confusing.

**Comment by madisonmay** [comment]
A few tiny comments and I think we're good to go!

**Comment by anneCarlson** [comment]
@madisonmay Made the updates; let me know when good to merge!

**Pull request merged**